app/src/main/java/at/bitfire/icsdroid/ui/InfoActivity.kt

```diff
@@ -43,30 +44,54 @@ import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.core.graphics.drawable.toBitmap
 import androidx.core.text.HtmlCompat
+import androidx.datastore.preferences.core.edit
 import at.bitfire.icsdroid.BuildConfig
 import at.bitfire.icsdroid.Constants
 import at.bitfire.icsdroid.R
+import at.bitfire.icsdroid.Settings.Companion.PrefNextReminder
+import at.bitfire.icsdroid.dataStore
+import at.bitfire.icsdroid.service.ComposableStartupService
+import at.bitfire.icsdroid.service.ComposableStartupService.Companion.FLAG_DONATION_DIALOG
 import at.bitfire.icsdroid.ui.partials.ExtendedTopAppBar
 import at.bitfire.icsdroid.ui.partials.GenericAlertDialog
 import at.bitfire.icsdroid.ui.theme.setContentThemed
 import com.mikepenz.aboutlibraries.ui.compose.LibrariesContainer
 import com.mikepenz.aboutlibraries.ui.compose.LibraryDefaults
+import kotlinx.coroutines.runBlocking
+import java.util.ServiceLoader
 
-class InfoActivity: ComponentActivity() {
+class InfoActivity: AppCompatActivity() {
+
+    /**
+     * Will get updated when the activity is created.
+     * Stores whether the donate dialog should be shown or not.
+     */
+    private var hasDonateDialogService: Boolean = false
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        // Init and collect all ComposableStartupServices
+        val compStartupServices = ServiceLoader.load(ComposableStartupService::class.java)
+            .also { srv ->
+                hasDonateDialogService = srv.any { it.flags or FLAG_DONATION_DIALOG == FLAG_DONATION_DIALOG }
```

Comment:
What is this? Could it not just be `it.flags == FLAG_DONATION_DIALOG`, or do I misunderstand something here?

Reply:
It could be, but if we ever add more flags for any reason, though I don't think so, that should be the way to go.

Reply:
Ah, yes. Maybe add some clarifying parenthesis. It took me a while to figure out what it means :D

Like this `(it.flags and FLAG_DONATION_DIALOG) == FLAG_DONATION_DIALOG`

Reply:
Or maybe even add a method to the ComposableStartupInterface:

```
fun hasFlag(flag: Int) = (flags and flag) == flag
```

So we can simply do this:

```
hasDonateDialogService = srv.any { it.hasFlag(FLAG_DONATION_DIALOG) }
```